beacon-chain/db/kv/attestations.go

```diff
@@ -113,6 +112,33 @@ func (k *Store) DeleteAttestation(ctx context.Context, attDataRoot [32]byte) err
 	})
 }
 
+// DeleteAttestations by attestation data roots.
+func (k *Store) DeleteAttestations(ctx context.Context, attDataRoots [][32]byte) error {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.DeleteAttestations")
+	defer span.End()
+	return k.db.Batch(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(attestationsBucket)
+		for _, r := range attDataRoots {
+			enc := bkt.Get(r[:])
+			if enc == nil {
+				return nil
+			}
+			att := &ethpb.Attestation{}
+			if err := proto.Unmarshal(enc, att); err != nil {
+				return err
+			}
+			indicesByBucket := createAttestationIndicesFromData(att.Data, tx)
+			if err := deleteValueForIndices(indicesByBucket, r[:], tx); err != nil {
+				return errors.Wrap(err, "could not delete root for DB indices")
+			}
+			if err := bkt.Delete(r[:]); err != nil {
```

Comment:
from my experience with batching in bolt if you don't call delete and get on a separate goroutine  batching becomes sequential and run the same as not batched requests to the db.
[Batch is only useful when there are multiple goroutines calling it.](https://godoc.org/github.com/boltdb/bolt#DB.Batch)

Reply:
+1, with how bolt works this will be just sequential and there would be no advantage of using `Batch` versus `Update`